### ISSUE TITLE
Presentation mode (docs to slides)

### ISSUE THREAD

**Issue by virgile-dev** [suitenumerique/docs#417]
## Feature Request

**Is your feature request related to a problem or unsupported use case? Please describe.**
Some users of the pad used the reveal.js support to make presentations.
This a geeky feature but it really aligns with our values of putting content first.
With reveal.js you make presentation using only text and don't do any formatting.

**Describe the solution you'd like**
Create BlockNote plugin that makes it compatible with reveal.js

**Comment by virgile-dev** [comment]
Today i was thinking about how we could implement this. 
We could create a custom block, something like the block code where you can write markdown and html.
The block could have 2 buttons : preview / edit  and another one for full screen.

**Comment by virgile-dev** [comment]
It could be a blockcode option like it's done for mermaid.js in outline
![Image](https://github.com/user-attachments/assets/9e3c4c2f-3384-408a-b42d-e2ff4fbdbecd)
Ping @YousefED

**Comment by YousefED** [comment]
> ## Feature Request
> **Is your feature request related to a problem or unsupported use case? Please describe.** Some users of the pad used the reveal.js support to make presentations. This a geeky feature but it really aligns with our values of putting content first. With reveal.js you make presentation using only text and don't do any formatting.
> 
> **Describe the solution you'd like** Create BlockNote plugin that makes it compatible with reveal.js

Interesting! Will give this some thought. Was reveal.js directly integrated in the pad? Or do you have examples of other applications where it's integrated well?

**Comment by virgile-dev** [comment]
Yes it's integrated directly. You type in your content and have a menu opens a presentation view.

**Comment by lsmith77** [comment]
I also like this idea.

I also like Docs to stay in a disruptive mode:
One of the things that is annoying with presentations vs. documentation is that the content overlaps. This means that there is a huge risk of content getting out of sync and redundant efforts to maintain the content. At the same time, the use cases are different, so the content should indeed be shown differently. 

I don't have a solution for this but it would be amazing if somehow one could pull together a presentation from documentation inside Docs without having to duplicate the content. more like applying a filter to make the content manageable during a presentation, but then, for example, being able to seamlessly dive deeper into audience questions require more depth.

**Comment by rasos** [comment]
A slide mode would be great for docs, just like hedgedoc does it with reveal.js see https://demo.hedgedoc.org/slide-example#

**Comment by virgile-dev** [comment]
Notion just released its [presentation mode](https://x.com/colebemis/status/2024543828372574422)
They went with something simpler than reveal.js which will be used by more users because it uses the directly the block strcuture of a document (not a code block with dark background).  Users just have to add seperator to indicate they want a new slide. They can enter the presentation mode using a sub-menu or ctrl + alt  + p which is also easy to exit. One drawback I see is you don't get a link to share your doc in presentation mode.
By default the presentation mode puts your browser in fullscreen on the tab (which is smart), when you move your mouse you get a small centered menu that allow you to exit full screen / presentation mode and a slide counter with arrow to navigate your content. The presentation mode works perfectly with keyboard arrows, I also suspect it works speaker remotes.
Here are a couple screenshot and [example notion page](https://www.notion.so/pyronear/LaSuite-LocalFirst-310425b63668803c935dccaf33c46092?source=copy_link) (use ctrl + alt  + p to trigger the presentation mode). 

<img width="2256" height="1504" alt="Image" src="https://github.com/user-attachments/assets/80c02938-fdac-4cb5-81c8-f160bdc6c088" />
<img width="2256" height="1504" alt="Image" src="https://github.com/user-attachments/assets/025503cb-63cc-450b-92d3-d29f65e09aba" />
<img width="2256" height="1504" alt="Image" src="https://github.com/user-attachments/assets/368153bd-b11f-4f81-9c46-da6c36bda8c9" />
<img width="2256" height="1504" alt="Image" src="https://github.com/user-attachments/assets/ea3a2bbe-7561-4d85-be15-5e82b9797f7d" />
<img width="1684" height="1309" alt="Image" src="https://github.com/user-attachments/assets/3b1edacc-80e5-4452-9cae-d0c1a6ea8084" />

**Comment by robin-lecomte** [comment]
[Here is my proposal](https://www.figma.com/design/qdCWR4tTUr7vQSecEjCyqO/Docs?node-id=13320-29459&t=3R0bcgyyeEuVLsy7-1)

Love how Notion handles this, it really reminds me of how [Craft](https://www.craft.do/fr) has been doing it for a few years now.  

Here’s my proposal:  
- Presentation mode that mimics slides  
- Variant of the main blocks to adapt to the larger area of the presenter mode (larger text and margins)
- New “Present” button to launch presentation mode  
- Redesigned and unified separator (I suggest keeping a single separator, like Notion does, to avoid user confusion, along with a refreshed design).

Looking forward to iterating on the topic!

<img width="3618" height="2116" alt="Image" src="https://github.com/user-attachments/assets/c4940edb-00db-4a67-9d70-e161d7cca75b" />

<img width="3630" height="2084" alt="Image" src="https://github.com/user-attachments/assets/126a2d15-7b6e-47bf-bfe1-d6115fa31673" />

**Comment by virgile-dev** [comment]
Hey @robin-lecomte this looks great !
I have almost no feedback except removing the page break block.
This was requested by user so asked BlockNote to build it.
While I agree it more of a form over content feature we should be careful when removing it. We could feature flag it and see what's the feedback form our user base.

A few question, that are popping to my mind: 
- Should the menu option to trigger the presentation mode be available to readers / anonymous visitors ? I feel this would help us avoid having a dedicated link for this
- This feels like a desktop only feature. Am I right ? Or should we plan on display a modal horizontaly when triggered on mobile ?

**Comment by kosssi** [comment]
❤️ It's crazy !

PS : [Outline release this feature](https://github.com/outline/outline/releases/tag/v1.6.0) 2 weeks ago ;)

**Comment by virgile-dev** [comment]
with @robin-lecomte we agreed today on removing the page break block. 
It's a formating feature that belongs in more advanced formatting feature like Libre Office or Word.
Users can still create a page break after they exported.
